h5py/_locks.pxi

```diff
@@ -111,7 +111,7 @@ cdef bint _acquire_lock(FastRLock lock, long current_thread, int wait) nogil:
     lock._count = 1
     return 1
 
-cdef inline void unlock_lock(FastRLock lock) nogil:
+cdef inline void unlock_lock(FastRLock lock) noexcept nogil:
```

Comment:
Should `lock_lock` and `_acquire_lock` above be changed the same way as well?

(Reposted on the relevant line)

Reply:
Hm good question. I noticed that Cython didn't complain about them so I didn't worry about it initially. I wonder if `bint` functions are treated differently by the except/noexcept checks. Maybe I'll try adding it locally and see how it goes. I should also check the generated C and see if it is doing anything special (ex. acquiring the GIL during an error check or something).

Reply:
Thanks! In that case, let's add `noexcept` to them explicitly so it's clear they're aligned with `unlock_lock`